### ISSUE TITLE
XSimGCL的均匀性

### ISSUE THREAD

**Issue by 404creep** [Coder-Yu/SELFRec#34]
我在对uniformity loss复现时发现uniformity loss趋势和论文中展示的差不多，数值不太一样，不知道是因为我的uniformity loss实现和你不一样或者是我对你的采样策略理解有误，方便公布一下代码吗

**Comment by Coder-Yu** [comment]
这个代码真没保存😅 趋势是一样就行

**Comment by 404creep** [comment]
好的，谢谢

**Comment by 404creep** [comment]
> > 
> 
> 您好！想请问下您复现uniformity loss的代码方便分享一下吗 谢谢

`def alignment(x, y):
    x, y = F.normalize(x, dim=-1), F.normalize(y, dim=-1)
    return (x - y).norm(p=2, dim=1).pow(2).mean()

def uniformity(x, t=2):
    x = F.normalize(x, dim=-1)
    return torch.pdist(x, p=2).pow(2).mul(-t).exp().mean().log()`

**Comment by forclh** [comment]
> 我在对uniformity loss复现时发现uniformity loss趋势和论文中展示的差不多，数值不太一样，不知道是因为我的uniformity loss实现和你不一样或者是我对你的采样策略理解有误，方便公布一下代码吗

您好可以分享一下uniformity loss实现的代码吗，感激不尽！forclh@qq.com

**Comment by 404creep** [comment]
> > 我在对uniformity loss复现时发现uniformity loss趋势和论文中展示的差不多，数值不太一样，不知道是因为我的uniformity loss实现和你不一样或者是我对你的采样策略理解有误，方便公布一下代码吗
> 
> 您好可以分享一下uniformity loss实现的代码吗，感激不尽！[forclh@qq.com](mailto:forclh@qq.com)
你可以参考SELFRec\util\loss_torch.py中的uniformity函数，具体我当时的代码我没有保留
